backend/windmill-api/src/job_metrics.rs

```diff
@@ -137,6 +150,59 @@ async fn get_job_metrics(
     let response = JobStatsResponse { metrics_metadata, scalar_metrics, timeseries_metrics };
     Ok(Json(response))
 }
+#[derive(Deserialize)]
+struct JobProgressSetRequest {
+    percent: i32,
+    flow_job_id: Option<Uuid>,
+}
+
+async fn set_job_progress(
+    Extension(db): Extension<DB>,
+    Path((w_id, job_id)): Path<(String, Uuid)>,
+    Json(JobProgressSetRequest { percent, flow_job_id }): Json<JobProgressSetRequest>,
+) -> error::JsonResult<()> {
+    dbg!(flow_job_id);
+
+    if let Some(flow_job_id) = flow_job_id {
+        sqlx::query!(
+            "UPDATE queue
+                SET flow_status = JSONB_SET(flow_status, ARRAY['modules', flow_status->>'step', 'progress'], $1)
+                WHERE id = $2",
+            serde_json::json!(percent.clamp(0, 99)),
+            flow_job_id
+        )
+        .execute(&db)
+        .await?;
+    }
+
+    record_metric(
+        &db,
+        w_id,
+        job_id,
+        "progress_perc".to_owned(),
+        MetricNumericValue::Integer(percent),
+    )
+    .await?;
+
+    Ok(Json(()))
+}
+
+async fn get_job_progress(
+    Extension(db): Extension<DB>,
+    Path((w_id, job_id)): Path<(String, Uuid)>,
+) -> error::JsonResult<Option<i32>> {
+    let progress: (Option<i32>, ) = sqlx::query_as(
+        //                    ^
+        // Trailing comma is mandatory, otherwise sqlx doesnt recognize it as a right type
```

Comment:
what about using query_scalar! like in api/jobs.rs?

Reply:
Fixed